### DISCUSSION TITLE
Run PostgREST in Raspberry Pi armv7

### DISCUSSION THREAD

**Discussion by hannesa2** [PostgREST/postgrest#3080] [Q&A] [answered]
I found here the only artifact for **armv7** [postgrest-v9.0.0-ubuntu-armv7.tar.xz](https://github.com/PostgREST/postgrest/releases/download/v9.0.0/postgrest-v9.0.0-ubuntu-armv7.tar.xz) in this release https://github.com/PostgREST/postgrest/releases/tag/v9.0.0

But during start, I run into this
<img width="744" alt="image" src="https://github.com/PostgREST/postgrest/assets/3314607/c9b30cd1-1a80-403e-bea3-02afe20157e5">

Any idea how to make it work in an Raspberry PI ?

<img width="591" alt="image" src="https://github.com/PostgREST/postgrest/assets/3314607/1ab11b36-48fc-4411-9b89-027062d5b7f9">

**Comment by hannesa2** [comment]
My paln was to to run these command

```
wget -4c https://ftp.gnu.org/gnu/glibc/glibc-2.29.tar.gz
tar -zxvf glibc-2.29.tar.gz
cd glibc-2.29
mkdir build_dir
cd build_dir
sudo ../configure --prefix=/opt/glibc
sudo make
sudo make install
```

but with 

    sudo ../configure --prefix=/opt/glibc

I run into

<img width="436" alt="image" src="https://github.com/PostgREST/postgrest/assets/3314607/9fd29bf7-c359-4850-907a-0bfc600367a9">

**Comment by hannesa2** [comment]
Or how to make it work on a Raspberry Pi ?

**Reply by laurenceisla** [reply]
The [Haskell Wiki](https://wiki.haskell.org/Raspberry_Pi) gives a link to this README on how to run Haskell software on a Raspberry Pi: https://github.com/blitzcode/hue-dashboard/blob/master/README.md#raspberry-pi

You could try with that. There's also a [v9.0.0 Docker image](https://hub.docker.com/layers/postgrest/postgrest/v9.0.0/images/sha256-564c94132abf861adfd46a9473b90e4cf06881814799490660022e1c780eaa02?context=explore) for armv7 that could help.

**Reply by hannesa2** [reply]
I would naively wish that the CI would also work for the Raspberry.

What prevents a ready-to-use zip or docker container on a Raspberry ? Or how can I help ?
I've no clue about Haskell.

**Reply by wolfgangwalther** [reply, marked as answer]
> What prevents a ready-to-use zip or docker container on a Raspberry?

This could be possible once we solve #3281.

**Comment by hannesa2** [comment]
I see the `latest` isn't supported

### Latest 👎
<img width="564" alt="image" src="https://github.com/PostgREST/postgrest/assets/3314607/b0611e17-2c94-44f3-8d62-9d50f64b1e1a">

### v9.0.0 👍
But `docker pull postgrest/postgrest:v9.0.0.20220211` works 

<img width="570" alt="image" src="https://github.com/PostgREST/postgrest/assets/3314607/8010f455-62f5-40aa-9f49-0e92f229273c">

### Starting v9.0.0

Here with 

```
docker run --rm --net=host \
   -e PGRST_DB_URI="postgres://pi@localhost/postgres" \
   postgrest/postgrest:v9.0.0.20220211 
```

I run into

> clock_gettime: Operation not permitted

Does someone knows how to get rid of it ?

**Reply by hannesa2** [reply]
> clock_gettime: Operation not permitted

Add a `--privileged` solves this permission issue

```
docker run --rm --privileged --net=host \
   -e PGRST_DB_URI="postgres://pi@localhost/postgres" \
   postgrest/postgrest:v9.0.0.20220211 
```
